operations/configuration_management/Write_Ansible_Code_for_CFS.md

```diff
@@ -38,15 +38,67 @@ CFS will handle scaling up Ansible to run on many hosts, but there are still pla
 * See [Target Ansible Tasks for Image Customization](Target_Ansible_Tasks_for_Image_Customization.md) for more information on writing for image customization.
 
 ### Disable fact gathering
 
 * Turn off facts that are not needed in a playbook by setting `gather_facts: false`. If only a few facts are required, it is also possible to limit fact gathering by setting `gather_subset`.
 For more information on `gather_subset`, see the external [Ansible module setup](https://docs.ansible.com/ansible/latest/modules/setup_module.html) documentation.
-* Avoid importing playbooks in other playbooks.  This will trigger fact gathering for each imported playbook, potentially collecting the same information multiple times.
+* Reducing fact gathering time is especially important when importing multiple playbooks from a top level playbook.
+  Fact gathering will trigger for each imported playbook, potentially collecting the same information multiple times.
+  
+### `group_by` and `add_host`
+
+The `group_by` and `add_host` modules can both be used to dynamically generate new hosts groups for the Ansible inventory.  This is useful when for hosts that be grouped according to a common property, and when plays or large sections of plays are intended to only target a particular kind of host.  Examples of this might include grouping by operating system, hardware type, or a hardware property such as the presence of a GPU.
+
+`group_by` should by used when there are multiple named groups that hosts can be grouped by.  For more instance on using this module see See Ansible's [playbooks best practices](https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#handling-os-and-distro-differences) guide for more information.
+
+```yaml
+- name: group by OS
+  hosts: all
+  tasks:
+    - name: Classify hosts by OS
+      group_by:
+        key: os_{{ ansible_facts['distribution'] }}
+
+- name: centOS playbook
+  hosts: os_CentOS
+  roles:
+    ...
+
+```
+
+`add_host` is useful for cases where the property is true or false.  It allows users to create a new group consisting of only the hosts where the property is true.  See Ansible's documentation on [](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/add_host_module.html) for more information.
+
+```yaml
+- name: group by a sample variable
+  hosts: all
+  tasks:
+    - name: Add all where sample_var is true to the new Sample group
+      add_host:
+        name: '{{ inventory_hostname }}'
+        groups: sample_group
+      when: sample_var
+
+- name: Sample playbook
+  hosts: sample_group
+  roles:
+    ...
+```
+
+To target only a new group, plays should use the following syntax.  In this example the play is targeting only images for `Compute` nodes. `&` takes the intersection of the `Compute` and `sample_group` groups.
```

Comment:
I feel like this sentence is a little misleading because your earlier example shows targeting only a new group, the sample_group. The examples that follows targets the intersection of two groups.
Again, I would italicize 'images' to highlight it.